core/src/test/java/org/apache/iceberg/jdbc/TestJdbcCatalog.java

```diff
@@ -161,6 +163,40 @@ public void testInitialize() {
     jdbcCatalog.initialize("test_jdbc_catalog", properties);
   }
 
+  @Test
+  public void testInitSchemaV0() throws Exception {
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put(CatalogProperties.WAREHOUSE_LOCATION, this.tableDir.toAbsolutePath().toString());
+    properties.put(CatalogProperties.URI, "jdbc:sqlite:file::memory:?icebergDBV0");
+    properties.put(JdbcUtil.SCHEMA_VERSION_PROPERTY, JdbcUtil.SchemaVersion.V0.name());
+    JdbcCatalog jdbcCatalog = new JdbcCatalog();
+    jdbcCatalog.setConf(conf);
+    jdbcCatalog.initialize("v0catalog", properties);
+
+    TableIdentifier tableIdent = TableIdentifier.of("db", "ns1", "ns2", "tbl");
+    Table table =
+        jdbcCatalog
+            .buildTable(tableIdent, SCHEMA)
+            .withPartitionSpec(PARTITION_SPEC)
+            .withProperties(null)
+            .withProperty("key1", "value1")
+            .withProperties(ImmutableMap.of("key2", "value2"))
+            .create();
+
+    assertThat(table.schema()).hasToString(SCHEMA.toString());
+    assertThat(table.spec().fields()).hasSize(1);
+    assertThat(table.properties()).containsEntry("key1", "value1").containsEntry("key2", "value2");
```

Comment:
I would probably just add a `listTables()` check here, to make sure the created table can be found